### ISSUE TITLE
Concourse version: v4.1.0 slow

### ISSUE THREAD

**Issue by aterreno** [concourse/concourse#2604]
* Concourse version: v4.1.0 
* Deployment type BOSH (via https://github.com/EngineerBetter/concourse-up) 
* Infrastructure/IaaS: AWS
* Browser (if applicable): (Any, Chrome)
* Did this used to work: yes, the previous version was fine 

Hi there,
we've been using concourse 3.x for about a year without troubles, but since we upgraded to 4.x we found it very much slower. 

I am not an expert, so I might need some pointers on how to troubleshoot and investigate further. 
However, the current performance of the worker doesn't seem reasonable. 

![image](https://user-images.githubusercontent.com/76042/45752371-253ab000-bc0d-11e8-8daa-1e0807b06bf7.png)

This is an m4.large running on AWS, when idle, the CPU is stuck at 50%, when we got a few projects running we are always at 100%. 

Just to be complete on the report:
web is t2.small doing fine
bosh is t2.small doing fine 
db.t2.small seems also doing well, we've got 40~ connections open even when a job is not running
We've got about 36 projects

We are a small team (4 devs), it's quite rare to have more than 2 builds running at the same time, when we run 5 or 6 it's slow as hell (our builds are mostly node.js on docker / npm install / deploy tasks on aws, so disk and network IO)

**Comment by aterreno** [comment]
The guys at concourse-up / Engineer better helped me a little to troubleshoot this, so I am sharing what we've got so far. 

Here are my worker logs, machine is idle (no jobs running) but with CPU at min 40+% 

[concourse.worker.0-20180927-104150-734269313.zip](https://github.com/concourse/concourse/files/2423586/concourse.worker.0-20180927-104150-734269313.zip)

<img width="887" alt="screenshot 2018-09-26 at 19 05 29" src="https://user-images.githubusercontent.com/76042/46139078-7f5af700-c245-11e8-99e5-07375905f275.png">

<img width="1919" alt="iotop" src="https://user-images.githubusercontent.com/76042/46139082-82ee7e00-c245-11e8-89ba-2d332eb11b74.png">
<img width="1892" alt="htop" src="https://user-images.githubusercontent.com/76042/46139083-82ee7e00-c245-11e8-9ab5-6f33f9b05105.png">

**Comment by jama-pivotal** [comment]
@aterreno out of curiosity, what specific version of Concourse 3.x did you upgrade from? I know in 3.10.0 we switched the default fs driver back to `btrfs`, that's the only thing I could think of that might have caused a performance degradation

**Comment by aterreno** [comment]
Hi @jama-pivotal thanks for looking into this, we upgraded from 3.14.1, we do use https://github.com/EngineerBetter/concourse-up 

just to update this thread as well, things got better now, https://github.com/EngineerBetter/concourse-up/issues/69 

We are on 30~ % Avg cpu on idle right now... we've got 20~ pipelines, so it seems still hight, albeit acceptable compared to what we had before. 

<img width="894" alt="screenshot 2018-10-01 at 17 49 31" src="https://user-images.githubusercontent.com/76042/46302958-7306cf00-c5a2-11e8-9fe7-422e186475b2.png">

**Comment by loghen41** [comment]
I'm currently seeing some similar behavior, and am in the process of optimizing our concourse instance. Here is our setup:

platform/iaas: aws ec2
concourse implimentation: concourse web and worker run via docker
instance size m5.larger
compute resource for workers: all pipelines/jobs utilize codebuild for dynamic compute power
active pipelines: ~70
CPU idle: 70%
RDS DB idle: 12%


We noticed that for every job we add in, the CPU idle process continues to increase.  Earlier, I increased the default check interval for git to 30min, and noticed that the CPU idle percentage decreased, (at the time it was ~50% for ~30 pipelines, and dropped down to ~25%.  But the CPU idle has increased as the amount of pipelines have been added.

Is there anything else I can investigate into to debug this. I'm surprised at the base CPU idle process as I continue to scale and add pipelines to the instance.
